src/module_loader/inner_loader.rs

```diff
@@ -165,9 +165,13 @@ impl InnerRustyLoader {
         kind: deno_core::ResolutionKind,
     ) -> Result<ModuleSpecifier, ModuleLoaderError> {
         #[cfg(feature = "node_experimental")]
-        let referrer_specifier = referrer
-            .to_module_specifier(&self.cwd)
-            .map_err(|e| JsErrorBox::from_err(to_io_err(e)))?;
+        let referrer_specifier = if deno_core::specifier_has_uri_scheme(referrer) {

```

Comment:
This was needed to fix one of the module loader unit tests, deno seems to be storing module paths internally as file:/// urls rather than preserving the relative paths from source